cpp/src/io/comp/unsnap.cu

```diff
@@ -614,11 +614,7 @@ __global__ void __launch_bounds__(block_size)
   unsnap_state_s *s = &state_g;
   int strm_id       = blockIdx.x;
 
-  if (t < sizeof(gpu_inflate_input_s) / sizeof(uint32_t)) {
-    reinterpret_cast<uint32_t *>(&s->in)[t] =
-      reinterpret_cast<const uint32_t *>(&inputs[strm_id])[t];
-    __threadfence_block();
-  }
+  if (t == 0) { s->in = inputs[strm_id]; }
```

Comment:
I'm a bit lost. Does `t < sizeof(gpu_inflate_input_s) / sizeof(uint32_t)` == `t == 0` for all cases?

Reply:
the old code copied the structure in 32bit pieces using multiple threads. Since the struct is fairly small, I replaced this is simple copy using the thread zero.